DashAI/back/tasks/base_task.py

```diff
@@ -59,6 +60,13 @@ def add_compatible_component(
 
         cls.compatible_components[registry_for.__name__][component.__name__] = component
 
+    @classmethod
+    def get_schema(self) -> dict:
+        """
+        This method load the schema JSON file asocciated to the task.
+        """
+        return json.load(open(f"DashAI/back/tasks/tasks_schemas/{self.name}.json", "r"))
```

Comment:
Here you have to use a context manager to close the file at the end of the read operation.
Also, add a try/except to catch the case where the schema file is not found.

```suggestion
        try:
            with open(f"DashAI/back/tasks/tasks_schemas/{self.name}.json", "r") as f:
                schema = json.load(f)
            return schema
        except FileNotFoundError:
            logger.exception(
                (
                    f"Could not load the schema for {self.__name__} : File DashAI/back"
                    f"/tasks/tasks_schemas/{self.name}.json not found."
                )
            )
            return {}
```